src/options.rs

```diff
@@ -0,0 +1,284 @@
+        }
+    }
+}
+
+impl OptionsBuilder {
+    fn setter_enabled(&mut self, x: bool) -> &mut Self {
+        if self.setter_enabled.is_some() {
+            warn!("Setter enabled already defined as {:?}, new value is {:?}.",
+                self.setter_enabled, x);
+        }
+        self.setter_enabled = Some(x);
+        self
+    }
+
+    fn setter_pattern(&mut self, x: SetterPattern) -> &mut Self {
+        if self.setter_pattern.is_some() {
+            warn!("Setter pattern already defined as {:?}, new value is {:?}.",
+                self.setter_pattern, x);
+        }
+        self.setter_pattern = Some(x);
+        self
+    }
+
+    fn setter_public(&mut self, x: bool) -> &mut Self {
+        if self.setter_public.is_some() {
+            warn!("Setter visibility already defined as {:?}, new value is {:?}.",
+                self.setter_public, x);
+        }
+        self.setter_public = Some(x);
+        self
+    }
+
+    fn setter_implicit_options(&mut self, x: bool) -> &mut Self {
+        if self.setter_implicit_options.is_some() {
+            warn!("Setter implicit options already defined as {:?}, new value is {:?}.",
+                self.setter_implicit_options, x);
+        }
+        self.setter_implicit_options = Some(x);
+        self
+    }
+
+    fn getter_enabled(&mut self, x: bool) -> &mut Self {
+        if self.getter_enabled.is_some() {
+            warn!("Getter enabled already defined as {:?}, new value is {:?}.",
+                self.getter_enabled, x);
+        }
+        self.getter_enabled = Some(x);
+        self
+    }
+
+    fn parse_attribute(&mut self, attr: syn::Attribute) {
+        trace!("Parsing attribute {:?}.", attr);
+        if attr.style != syn::AttrStyle::Outer || attr.is_sugared_doc {
+            trace!("Ignoring attribute (outer or sugared doc).");
+            return
+        }
+
+        // e.g. `#[setters]`
+        if let syn::MetaItem::Word(ref ident) = attr.value {
+            match ident.as_ref() {
+                "setters" => {
+                        self.setter_enabled(true);
+                        return
+                    },
+                "getters" => {
+                        self.getter_enabled(true);
+                        return
+                    },
+                _ => {}
+            }
+        }
+
+        // e.g. `#[setters(...)]`
+        if let syn::MetaItem::List(ref ident, ref nested_attrs) = attr.value {
+            match ident.as_ref() {
+                "setters" => {
+                        self.setter_enabled(true);
+                        self.parse_setters_options(nested_attrs);
+                        return
+                    },
+                "getters" => {
+                        self.getter_enabled(true);
+                        self.parse_getters_options(nested_attrs);
+                        return
+                    },
+                _ => {}
+            }
+        }
+        debug!("Ignoring attribute.");
```

Comment:
I would not use these attribute names, as they live in a global namespace and "getters" and "setters" is too generic (we just had that in diesel/rocket with "options"). How about something like `#[builder_options(setters(pattern = mutable, private = true), getter(public = true)]`?